Audio PR 5 -Audio Augmentations 1

swetha097 opened the pull request:
- Add spectrogram support in the rocAL

Commit:
Add NSR and Spectrogram support in the rocAL

fiona-gladwin committed:
Merge branch 'audio_pr4' into swbs/audio/pr5

fiona-gladwin committed:
Fix output enum in pybind

fiona-gladwin committed:
Change type of NSR outputs wrt RPP TOT

fiona-gladwin committed:
Minor fix - num threads in audio decoder

snehaa8 reviewed (commented):
@swetha097 Please address review comments and also recheck the PR for other such minor issues.

swetha097 committed:
Merge remote-tracking branch 'origin/audio_pr4' into swbs/audio/pr5

swetha097 committed:
Remove NSR

Commit:
Minor additions in the .h file

Commit:
Merge remote-tracking branch 'origin/audio_pr4' into swbs/audio/pr5

Commit:
Formatting Changes

Commit:
Formatting , review comments resolution and change enum dtype to int32 instead of uint32

swetha097 committed:
Merge audio_pr4 branch

Commit:
Remove the reset_tensor_roi calls

Commit:
Resolve some PR comments

Commit:
Merge branch 'audio_pr4' into swbs/audio/pr5

Commit:
Minor changes

Commit:
Change the dims[0] and dims[1] positioning for Spectrogram

swetha097 committed:
Resolving PR comments

swetha097 committed:
Merge remote-tracking branch 'swe_fork/audio_pr4' into swbs/audio/pr5

Commit:
Minor Change

Commit:
Minor change

swetha097 committed:
Revert "Change the dims[0] and dims[1] positioning for Spectrogram"

This reverts commit d791b9ac77d662f1528088165d2cfdba721a2b4f.

swetha097 committed:
Add Spectrogram Case in unit tests

swetha097 committed:
Merge branch 'swbs/audio/pr5' of https://github.com/swetha097/rocAL into swbs/audio/pr5

swetha097 committed:
Add spectrogram case in python unit tests

swetha097 committed:
Fix review comments

swetha097 committed:
Merge remote-tracking branch 'swe_fork/audio_pr4' into swbs/audio/pr5

SundarRajan28 committed:
Adding missed param in python unit tests

swetha097 committed:
Update ChangeLog.md

swetha097 committed:
Merge branch 'swbs/audio/pr5' of https://github.com/swetha097/rocAL into swbs/audio/pr5

SundarRajan28 committed:
Merge branch 'audio_pr4' into swbs/audio/pr5

SundarRajan28 committed:
Merge branch 'audio_pr4' into swbs/audio/pr5

SundarRajan28 committed:
Merge branch 'audio_pr4' into swbs/audio/pr5

swetha097 committed:
Fix build issue

fiona-gladwin committed:
Introduce Audio layouts

fiona-gladwin committed:
Add layout changes for spectrogram

swetha097 committed:
Fix the unit tests - c++ & python

fiona-gladwin committed:
Merge branch 'swbs/audio/pr5' of https://github.com/swetha097/rocAL into swbs/audio/pr5_layout

fiona-gladwin committed:
Minor fix

fiona-gladwin committed:
Merge branch 'swbs/audio/pr5_layout' into swbs/audio/pr5

kiritigowda committed:
Readme - Updates (#153)

* Readme - Updates

* Setup - Cleanup

* CMakeLists - remove OBSOLETES

* Readme - updates

* Updates

swetha097 committed:
Merge branch 'audio_pr4' into swbs/audio/pr5

samjwu committed:
Docs - Update Read the Docs configuration to use Python 3.10 and latest rocm-docs-core (#158)

dependabot[bot] committed:
Docs - Bump rocm-docs-core[api_reference] from 1.2.0 to 1.2.1 in /docs/sphinx (#159)

Bumps [rocm-docs-core[api_reference]](https://github.com/RadeonOpenCompute/rocm-docs-core) from 1.2.0 to 1.2.1.
- [Release notes](https://github.com/RadeonOpenCompute/rocm-docs-core/releases)
- [Changelog](https://github.com/ROCm/rocm-docs-core/blob/develop/CHANGELOG.md)
- [Commits](https://github.com/RadeonOpenCompute/rocm-docs-core/compare/v1.2.0...v1.2.1)

---
updated-dependencies:
- dependency-name: rocm-docs-core[api_reference]
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

LakshmiKumar23 committed:
Audio - Version check  (#156)

* version check from MIVisionX vx_rpp for Audio

* check if version file exists

* check for vx_rpp version and sndfile before settign rocal_audio

* remove unnecessary find_pacakge

* check for sndfile only if vx_rpp supports audio

dependabot[bot] committed:
Docs - Bump rocm-docs-core[api_reference] from 1.2.1 to 1.3.0 in /docs/sphinx (#163)

Bumps [rocm-docs-core[api_reference]](https://github.com/RadeonOpenCompute/rocm-docs-core) from 1.2.1 to 1.3.0.
- [Release notes](https://github.com/RadeonOpenCompute/rocm-docs-core/releases)
- [Changelog](https://github.com/ROCm/rocm-docs-core/blob/develop/CHANGELOG.md)
- [Commits](https://github.com/RadeonOpenCompute/rocm-docs-core/compare/v1.2.1...v1.3.0)

---
updated-dependencies:
- dependency-name: rocm-docs-core[api_reference]
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

kiritigowda committed:
Setup - Updates (#164)

* Setup - Updates

* Setup - pip packages updates

* CMakeList - Add Protobuf deps

swetha097 committed:
Audio PR - Augmentation support [PreEmphasis Filter] (#122)

* Audio Decoder PR 1

* channge image_info to sample_info to maintain a generic name for all the use-cases

* Change the copyright year from 2023 to 2024

* formatting the files

* Resolve PR comments

* NWC

* Resolve PR comments

* Change decoded_img_info to decoded_video_info

* Change the file_path() function to virtual from pure virtual

* Minor change

* Minor changes

* Add the unit test file

* Revert "Add the unit test file"

This reverts commit e79cc06d3c9d167dc8d142d315b71a5bd5e86736.

* Fix the ROI api in python end

* Fix the ROI api in python end

* Remove the ROI change from pytorch.py

* Add Python Unit Test for Audio

* PreEmphasis Filter

* Add Preemp filter augmentations files

* Introduce CMake for sndfile

Modify CMakeLists.txt for the same

* Resolve 1st set of PR commenst

* Remove commented code for last batch polices and unsued imports

* ROI related changes - change from xy to wh to use for samples and channels

* Fix seg fault with ROI

* Remove opencv usage from the unit test

* Resolve the PR comments

* Remove instances of the audio_*_time - use the existing variables from Timing struct

* Formatting changes in rocal_api_data_loader.cpp and add the opencl and hip conditions for audio loaders

* Resolve the internal PR comments

* Reformatting the file_source_reader.cpp

* Remove _input_path from audio_source_evaluator and audio_read_and_decode as it is unecessary

* Change the header formatting

* Changes in copy_data() for audio samples

* Initialize the status at the beginning

* Resolve PR comments

* Resolving PR comments

* Resolve PR comments

* Remove the unused variables

* Resolve PR commensts

* Cmake related changes for audio

* Resolve PR comments

* Add condition check to eliminate any other file extensions other than a wav file / other image formats and call open_folder deom subfolder_reading() function

* Add basic testing for Audio to compare Bin files

* Update audio_read_and_decode.cpp

* Update rocAL_api_audio_unittest.py

* Revert file source reader changes

* Update master_graph.cpp

* Update tensor.cpp - Remove a commented line of code

* Rephrase

* Introduce ROCAL_AUDIO flag

Introduce flag for audio code, to be disabled when sndfile not found

* Minor changes

* Minor changes

* Add output comparision for Audio outputs

* Minor changes

* Add comment

* Minor changes to unit test

* Remove max_frames and max_channels args

* Remove max_frames, max_channels and sample rate from unit test

* Remove the NONE types

* Clean up unit test

* Clean up code in audio python unit test

* Introduce ROI in the unit test

* Minor change

* Format python unit test

* Add python script to run audio unittests

* Fix formatting issues - Minor changes

* Clean up C++ audio unit test

* Remove max frames and channels from decoder

* Minor changes

* Add Output comparison for python audio unittests

* Modify rocal audio unit test

Update README

* Minor change

* Resolve PR comments

* Minor changes - Modifying the names of the arguments

* Added a bried desc for the rocAL enum for border type

* Add a WRN statement in PreEmphasis Filter to only use FP32 dtype

* minor changes

* Change the borderType enum to int32 from uint32 dtype

* Minor changes

* Minor change

* Update README for audio unit test

* Parameters for rocALAudioIterator

* Minor change

* Remove the reset_tensor_roi() from the PreEmphasis augmentation making sure the src and dst roi points to the same location

* Del the Unit Test Files introduced earlier

* Add pre_emphasis function and gollden output comparison in audio unit tests

* minor change - add update val in create array

* Minor variable name change

* minor change

* Update unit test

* Minor change

* minor change

* Minor changes

Add wav extension in file reader
Add reader in unit test

* Minor change

* Add file reader to python audio unit test

* Update C++ unit test

* Update python audio unit test

* Remove the unused variable output - Resolve warnings in cpp unit test

* Remove the dst_roi arg passed to rpp

* Minor changes

Change borderType enum prefix

* Name change from sample to data

* Change from decoded_data_info to DecodedDataInfo

* Remove audio_decoder_factory.cpp file

* Minor change

* Change variable name

* Update the struct variable name in audio files

* Minor changes

* Change ROCAL_DATA_PATH to exclude rocal_data

* Update ROCAL_DATA_PATH to exclude rocal_data

* Use Pascal case for function names in audio decoder

* Modify cmake to have SNDFILE in all capital

* Minor changes

* Add struct for audio info in AudioReadAndDecode

* Fix merge conflict

* Renaming crop_image_info to CropImageInfo

* Remove - actual_host_buffers - Unused

* Rename TimingDBG to TimingDbg

* Move the instances of DecodedDataInfo to its base class LoaderModule

* Fix a WRN msg in master_graph.cpp

* Remove a dangling comment

* Rename _circ_data_info to _circ_buff_data_info

* Add Glob to CMakeLists.txt

* Rename SndFileDecoder to GenericAudioDecoder

* Fix build issues

* Minor change

* Update python API README.md for audio unit test

* Update audio unit test README

* Revert "Add Glob to CMakeLists.txt"

This reverts commit 47263d94393e9614a5cebcab6c11c804ac7152f8.

* Fix include headers for Audio files

* Fix copy data 2D

* Minor changes

* Pass decoded data info to load routine instead of separate vectors

* Update CHANGELOG.md

* Update CHANGELOG.md

* Change swap_handle_time variable name in loader

* Update the changelog.md

* Formatting changes

Add comments

* Update doxygen comments

* Move file source reader from readers/image to readers folder

* Update README and add doxygen description

* Update CMakeLists and README for audio test

* Update README for audio test

* Minor fix

* Fix merge from PR 2

* Minor changes shard_count argument name

* Rename set and get functions of data_info to decoded_data_info

* Revert empty line removed in CMakeLists.txt

* Removed prefix original for audio vectors

* Resolve PR comments

* Add @params to all args in pytorch.py

* Minor changes in unit test

* Minor changes

* Change ROCAL instaces to rocAL in pytorch.py

* Resolve the PR comments

* Minor changes in decoders.py - Modify the comment for shard_size

* Minor changes

* Address the PR comments

* Address Review comments

---------

Co-authored-by: root <root@ixt-sjc2-54.local.lan>
Co-authored-by: Swetha B S <swetha@multicorwareinc.com>
Co-authored-by: fiona-gladwin <fionagladwin@multicorewareinc.com>
Co-authored-by: Fiona-MCW <70996026+fiona-gladwin@users.noreply.github.com>
Co-authored-by: Swetha B S <>
Co-authored-by: SundarRajan28 <sundarrajan@multicorewareinc.com>
Co-authored-by: Kiriti Gowda <kiritigowda@gmail.com>

kiritigowda committed:
Readme - Updates (#165)

dependabot[bot] committed:
Bump rocm-docs-core[api_reference] from 1.3.0 to 1.4.0 in /docs/sphinx (#166)

Bumps [rocm-docs-core[api_reference]](https://github.com/ROCm/rocm-docs-core) from 1.3.0 to 1.4.0.
- [Release notes](https://github.com/ROCm/rocm-docs-core/releases)
- [Changelog](https://github.com/ROCm/rocm-docs-core/blob/develop/CHANGELOG.md)
- [Commits](https://github.com/ROCm/rocm-docs-core/compare/v1.3.0...v1.4.0)

---
updated-dependencies:
- dependency-name: rocm-docs-core[api_reference]
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

SundarRajan98 committed:
Audio PR - Adding support for file list reader (#123)

* Introduce CMake for sndfile

Modify CMakeLists.txt for the same

* Adding file list reader to audio python example

* Fixing reader import issue

* Resolve 1st set of PR commenst

* Remove commented code for last batch polices and unsued imports

* ROI related changes - change from xy to wh to use for samples and channels

* Fix seg fault with ROI

* Remove opencv usage from the unit test

* Resolve the PR comments

* Remove instances of the audio_*_time - use the existing variables from Timing struct

* Formatting changes in rocal_api_data_loader.cpp and add the opencl and hip conditions for audio loaders

* Updating python example

* Resolve the internal PR comments

* Reformatting the file_source_reader.cpp

* Remove _input_path from audio_source_evaluator and audio_read_and_decode as it is unecessary

* Change the header formatting

* Fixing issues with file list reader

* Changes in copy_data() for audio samples

* Initialize the status at the beginning

* Resolving review comments

* Resolve PR comments

* Resolving PR comments

* Resolve PR comments

* Remove the unused variables

* Resolve PR commensts

* Cmake related changes for audio

* Resolve PR comments

* Add condition check to eliminate any other file extensions other than a wav file / other image formats and call open_folder deom subfolder_reading() function

* Add basic testing for Audio to compare Bin files

* Removing storage_type from audio loader API

* Update audio_read_and_decode.cpp

* Update rocAL_api_audio_unittest.py

* Revert file source reader changes

* Update master_graph.cpp

* Update tensor.cpp - Remove a commented line of code

* Rephrase

* Introduce ROCAL_AUDIO flag

Introduce flag for audio code, to be disabled when sndfile not found

* Minor changes

* Minor changes

* Add output comparision for Audio outputs

* Minor changes

* Add comment

* Minor changes to unit test

* Remove max_frames and max_channels args

* Remove max_frames, max_channels and sample rate from unit test

* Remove the NONE types

* Clean up unit test

* Clean up code in audio python unit test

* Introduce ROI in the unit test

* Minor change

* Format python unit test

* Add python script to run audio unittests

* Fix formatting issues - Minor changes

* Clean up C++ audio unit test

* Remove max frames and channels from decoder

* Minor changes

* Add Output comparison for python audio unittests

* Modify rocal audio unit test

Update README

* Minor change

* Resolve PR comments

* Minor changes - Modifying the names of the arguments

* Initial commit for removing file list reader

* Added a bried desc for the rocAL enum for border type

* Add a WRN statement in PreEmphasis Filter to only use FP32 dtype

* minor changes

* Change the borderType enum to int32 from uint32 dtype

* Minor changes

* Minor change

* Update README for audio unit test

* Parameters for rocALAudioIterator

* Removing file list reader and metadata reader

* Minor change

* Remove the reset_tensor_roi() from the PreEmphasis augmentation making sure the src and dst roi points to the same location

* Del the Unit Test Files introduced earlier

* Changing python unittests for QA mode

* Resolving review comments

* Adding comment for file list case in file source reader

* Add pre_emphasis function and gollden output comparison in audio unit tests

* minor change - add update val in create array

* Minor variable name change

* minor change

* Update unit test

* Minor change

* minor change

* Resolving review comments

* Minor changes

Add wav extension in file reader
Add reader in unit test

* Minor change

* Add file reader to python audio unit test

* Update C++ unit test

* Update python audio unit test

* Remove the unused variable output - Resolve warnings in cpp unit test

* Remove the dst_roi arg passed to rpp

* Minor changes

Change borderType enum prefix

* Adding file list reader to C++ unit tests

* Fixing issues with C++ audio unit tests

* Resolving review comments

* Resolving review comments

* Updating audio unit tests for default file list path

* Minor changes

* Name change from sample to data

* Change from decoded_data_info to DecodedDataInfo

* Remove audio_decoder_factory.cpp file

* Minor change

* Change variable name

* Update the struct variable name in audio files

* Minor changes

* Change ROCAL_DATA_PATH to exclude rocal_data

* Update ROCAL_DATA_PATH to exclude rocal_data

* Use Pascal case for function names in audio decoder

* Modify cmake to have SNDFILE in all capital

* Minor changes

* Add struct for audio info in AudioReadAndDecode

* Fix merge conflict

* Renaming crop_image_info to CropImageInfo

* Remove - actual_host_buffers - Unused

* Rename TimingDBG to TimingDbg

* Move the instances of DecodedDataInfo to its base class LoaderModule

* Fix a WRN msg in master_graph.cpp

* Remove a dangling comment

* Rename _circ_data_info to _circ_buff_data_info

* Add Glob to CMakeLists.txt

* Rename SndFileDecoder to GenericAudioDecoder

* Fix build issues

* Minor change

* Update python API README.md for audio unit test

* Update audio unit test README

* Revert "Add Glob to CMakeLists.txt"

This reverts commit 47263d94393e9614a5cebcab6c11c804ac7152f8.

* Fix include headers for Audio files

* Fix copy data 2D

* Minor changes

* Pass decoded data info to load routine instead of separate vectors

* Update CHANGELOG.md

* Update CHANGELOG.md

* Change swap_handle_time variable name in loader

* Update the changelog.md

* Update CHANGELOG.md

* Formatting changes

Add comments

* Update doxygen comments

* Move file source reader from readers/image to readers folder

* Update README and add doxygen description

* Update CMakeLists and README for audio test

* Update README for audio test

* Minor fix

* Fix merge from PR 2

* Minor changes shard_count argument name

* Rename set and get functions of data_info to decoded_data_info

* Revert empty line removed in CMakeLists.txt

* Removed prefix original for audio vectors

* Resolve PR comments

* Add @params to all args in pytorch.py

* Minor changes in unit test

* Minor changes

* Change ROCAL instaces to rocAL in pytorch.py

* Resolve the PR comments

* Minor changes in decoders.py - Modify the comment for shard_size

* Minor changes

* Address the PR comments

* Address Review comments

* Resolving review comments

* Removing file_list_path from label_reader

---------

Co-authored-by: fiona-gladwin <fionagladwin@multicorewareinc.com>
Co-authored-by: SundarRajan28 <sundarrajan@multicorewareinc.com>
Co-authored-by: Swetha B S <swetha@multicorwareinc.com>
Co-authored-by: swetha097 <59434434+swetha097@users.noreply.github.com>
Co-authored-by: Fiona-MCW <70996026+fiona-gladwin@users.noreply.github.com>
Co-authored-by: swetha097 <swetha@multicorewareinc.com>
Co-authored-by: Swetha B S <>

fiona-gladwin committed:
Merge branch 'develop' of https://github.com/ROCm/rocAL into swbs/audio/pr5

fiona-gladwin committed:
Fix merge conflicts

fiona-gladwin committed:
Minor changes

fiona-gladwin committed:
Update README
